### ISSUE TITLE
Should I pass the untrainable model (which is used to extract features during training) to accelerator.prepare?

### ISSUE THREAD

**Issue by JWargrave** [huggingface/accelerate#1345]
Hi, there.

I am new to `accelerate` and I've found that it really improves my development productivity. Thanks for your great work.

But I have some problems when using `accelerator.prepare`.

Suppose the training process involves two models: `feature_extractor` and `trainable_model`.
The `feature_extractor` is used to extract features from training data and is kept frozen during training.
The `trainable_model` is the model we want to optimize.
The pseudocode of the training process is as follows:
```python
# suppose the training process involves two models:
# feature_extractor(untrainable),trainable_model(trainable)
for data,label in dataloader:
    ...
    with torch.no_grad():
        feature=feature_extractor.extract(data)
    loss=trainable_model(feature,label)
    ...
```

If I want to train on **multi GPUs**, should I wrap the `feature_extractor` with `accelerator.prepare`(i.e., `feature_extractor=accelerator.prepare(feature_extractor)`)?

**Comment by sgugger** [comment]
Normally it won't do anything to `feature_extractor` if it has no parameters with gradients, so passing it or not should be exactly the same.

**Comment by JWargrave** [comment]
> Normally it won't do anything to `feature_extractor` if it has no parameters with gradients, so passing it or not should be exactly the same.

ok,thanks a lot.
